commit 55ce1e65ec6fd4ed689646e22f8755e8658b35df
Author:     nvkevlu <55759229+nvkevlu@users.noreply.github.com>
AuthorDate: Wed Mar 23 12:22:21 2022 -0400
Commit:     GitHub <noreply@github.com>
CommitDate: Wed Mar 23 12:22:21 2022 -0400

    Initial checkin of code with job management and admin commands (#332)
    
    * initial checkin of code with job management with admin commands
    
    * make revisions to address comments in PR
    
    * Update __init__.py
    
    fix ci
    
    * Update file_transfer.py
    
    fix typo
    
    * fix ci
---
 nvflare/private/fed/server/server_engine.py | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/nvflare/private/fed/server/server_engine.py b/nvflare/private/fed/server/server_engine.py
index 6157087b..1573ecbe 100644
--- a/nvflare/private/fed/server/server_engine.py
+++ b/nvflare/private/fed/server/server_engine.py
@@ -31,9 +31,12 @@ from nvflare.apis.fl_constant import FLContextKey, MachineStatus, ReservedTopic,
 from nvflare.apis.fl_context import FLContext
 from nvflare.apis.fl_snapshot import FLSnapshot
 from nvflare.apis.shareable import Shareable, make_reply
+from nvflare.apis.study_manager_spec import StudyManagerSpec
 from nvflare.apis.utils.fl_context_utils import get_serializable_data
 from nvflare.apis.workspace import Workspace
+from nvflare.app_common.storages.filesystem_storage import FilesystemStorage
 from nvflare.fuel.hci.zip_utils import zip_directory_to_bytes
+from nvflare.mt.job_def_manager import SimpleJobDefManager
 from nvflare.private.admin_defs import Message
 from nvflare.private.fed.server.server_json_config import ServerJsonConfigurator
 from nvflare.widgets.info_collector import InfoCollector
@@ -78,6 +81,11 @@ class ServerEngine(ServerEngineInternalSpec):
         self.lock = Lock()
         self.logger = logging.getLogger(self.__class__.__name__)
 
+        # self.job_def_manager = job_def_manager  # todo: need to figure out how to initialize job manager with inputs
+        self.job_def_manager = SimpleJobDefManager(
+            StudyManagerSpec(), FilesystemStorage(root_dir="/workspace/nvflare_provis")
+        )
+
         self.asked_to_stop = False
         self.snapshot_persistor = snapshot_persistor
 
